### PR TITLE
Add basic components with styles and react router

### PR DESCRIPTION
**Pull request by natalix**
**What has been added?**

- Emotion for styling 
- Routing (react-router-dom)
- Icons library (react-icons)
- Basic structure of components with styles, Recipe page base structure
- Global styles with ThemeProvider
- Breakpoints

### CODE REVIEW

File: src/components/favourites/index.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { BsHeart } from 'react-icons/bs';
+
+const Favourites:React.FC = () => {
+
+  return (
+    <>
```
Comment (codejet) [inline src/components/favourites/index.tsx:7]: minor, but no need need to wrap a single component with a fragment (maybe this file was generated?)

File: src/components/favourites/index.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { BsHeart } from 'react-icons/bs';
+
+const Favourites:React.FC = () => {
```
Comment (codejet) [inline src/components/favourites/index.tsx:4]: we have a few occurrences at olx still, but we generally don't use React.FC. you can find some info in here: https://kentcdodds.com/blog/how-to-write-a-react-component-in-typescript

File: src/components/image-recipe/index.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  z-index: 1;
+  font-family: 'Merriweather',serif;
+  font-size: 30px;
+  font-weight: 300;
+  border-bottom: solid 1px #ffffff;
+  padding-bottom: 5px;
+  margin: 0 5px 10% 10%;
+  text-overflow: ellipsis;
+  overflow: hidden;
+  display: -webkit-box;
```
Comment (codejet) [inline src/components/image-recipe/index.tsx:30]: isn't that an outdated flexbox declaration? 🤔

File: src/components/index.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export * from './styles'
+export { default as Header } from './header'
```
Comment (codejet) [inline src/components/index.ts:2]: i would rather discourage reexports (https://twitter.com/iamakulov/status/1331551351214645251) even thought for this app it isn't a problem.

also, at olx we decided to favor named over default exports where possible. here is an article by one of our colleagues: https://blog.piotrnalepa.pl/2020/06/26/default-exports-vs-named-exports/

File: src/components/shopping-list/index.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { FaOpencart } from 'react-icons/fa';
+
+const ShoppingList:React.FC = () => {
+
+  return (
+    <>
```
Comment (codejet) [inline src/components/shopping-list/index.tsx:7]: see https://github.com/natalix/cook-book-app/pull/3/files?short_path=d4c9a8c#r657127416

Comment (codejet) [reply src/components/shopping-list/index.tsx:7]: actually, you are probably going to add more stuff here, in which case you might need the fragment at some point. just ignore the comment then.

Comment (natalix) [reply src/components/shopping-list/index.tsx:7]: yes, I was thinking that there should be some logic in here, that's why I used it

File: src/components/image-recipe/index.tsx (hunk 3 above)
Comment (natalix) [reply src/components/image-recipe/index.tsx:30]: If we want to have control over visible amount of lines via CSS (`-webkit-line-clamp: 3; `) we need to use it in combination with `display: -webkit-box`.
https://caniuse.com/?search=line-clamp

File: src/styles/global.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
+import { Global, css } from '@emotion/react';
+
+import { theme } from './'
+
+const GlobalStyles = () => {
+  return (
+    <Global
+    	styles={css`
+      /* RESET CSS */
```
[the remark this answers is not in the document]
Comment (natalix) [reply src/styles/global.tsx]: Thanks!

File: src/components/favourites/index.tsx (hunk 2 above)
Comment (natalix) [reply src/components/favourites/index.tsx]: I've removed all React.FC from previous commit, now it is more clear to me (and in fact there is no need to use at this moment).

File: src/components/index.ts (hunk 4 above)
Comment (natalix) [reply src/components/index.ts]: I've refactored all of exports.

**Pull request merged**